app/daemon.go

```diff
@@ -142,10 +142,7 @@ func startManager(c *cli.Context) error {
 
 	monitoring.InitMonitoringSystem(logger, currentNodeID, ds, kubeconfigPath)
 
-	if err := ds.InitSettings(); err != nil {
-		return err
-	}
-
+	// Initialize the required settings first
 	if err := updateDefaultImageSetting(m, types.SettingNameDefaultEngineImage, engineImage); err != nil {
```

Comment:
NIT: It's better to rename this function as well. e.g., `createOrUpdateDefaultImageSetting`

Reply:
Updated!